clang/lib/Sema/SemaExpr.cpp

```diff
@@ -11246,6 +11246,12 @@ static void DiagnoseBadShiftValues(Sema& S, ExprResult &LHS, ExprResult &RHS,
   if (S.getLangOpts().OpenCL)
     return;
 
+  if (LHS.get()->IgnoreParenImpCasts()->getType()->isBooleanType()) {
+    S.Diag(Loc, diag::warn_shift_bool)
+        << (Opc == BO_Shr) /*left|right*/ << LHS.get()->getSourceRange();
```

Comment:
Should we only warn on right shifts as the issue comments suggest?

Reply:
@Fznamznon do you mean omitting the handling of the right shift? I suppose both `<<` and `>>` convert the type to `int`., so handling both seem appropriate. Or does it not?

Reply:
Yeah, they both convert to int. However a right shift of a bool will often result in zero. A left shift though could have been intended. Weird but intended. 
I don't really have a strong opinion on this, but original comments from @zygoloid in https://github.com/llvm/llvm-project/issues/28334#issuecomment-980962562 suggests so. I wanted to consider all the options.

Reply:
Thanks. I agree that it’s important to discuss the `right shift` behavior before moving forward with these changes. If we decide to remove the right shift, I’ll update the PR.

Reply:
As depressing as this is, it is used in the wild:

https://sourcegraph.com/search?q=context:global+lang:C%2B%2B+%5B+%5Ct%5C%28%5Dtrue%5Cs*%3C%3C%5Cs*.*&patternType=regexp&sm=0

https://sourcegraph.com/search?q=context:global+lang:C+%5B+%5Ct%5C%28%5Dtrue%5Cs*%3C%3C%5Cs*.*&patternType=regexp&sm=0

Not super often, but enough that it might be considered too chatty to enable by default.

Reply:
@AaronBallman Yes, there are some cases related to left shift, however, I haven't found any for the right shift. Should we handle only the left shift?

Reply:
I think we should only diagnose right shift (`bool >> val`) for now and not diagnose left shift (`bool << val`).

Reply:
@AaronBallman could you please review the latest changes? thanks